tests/response/test_base_response.py

```diff
@@ -0,0 +1,178 @@
+
+
+def test_delete_cookie() -> None:
+    @get("/create")
+    def create_cookie_handler() -> Response:
+        response = Response(content=None)
+        response.set_cookie(
+            "test",
+            "abc",
+            max_age=60,
+            expires=60,
+            path="/",
+            secure=True,
+            httponly=True,
+            samesite="lax",
+        )
+        assert len(response.cookies) == 1
+        return response
+
+    @get("/delete")
+    def delete_cookie_handler() -> Response:
+        response = Response(content=None)
+        response.delete_cookie(
+            "test",
+            "abc",
+        )
+        assert len(response.cookies) == 1
+        return response
+
+    with create_test_client(route_handlers=[create_cookie_handler, delete_cookie_handler]) as client:
+        response = client.get("/create")
+        assert response.cookies.get("test") == "abc"
+        assert client.cookies.get("test") == "abc"
+        response = client.get("/delete")
+        assert response.cookies.get("test") is None
+        # the commented out assert fails, because of the starlette test client's behaviour - which doesn't clear
+        # cookies.
+        # assert client.cookies.get("test") is None
+
+
+@pytest.mark.parametrize(
+    "media_type, expected, should_have_content_length",
+    ((MediaType.TEXT, b"", False), (MediaType.HTML, b"", False), (MediaType.JSON, b"null", True)),
+)
+def test_empty_response(media_type: MediaType, expected: bytes, should_have_content_length: bool) -> None:
+    @get("/", media_type=media_type)
+    def handler() -> None:
+        return
+
+    with create_test_client(handler) as client:
+        response = client.get("/")
+        assert response.content == expected
+        if should_have_content_length:
+            assert "content-length" in response.headers
+        else:
+            assert "content-length" not in response.headers
+
+
+@pytest.mark.parametrize(
+    "status, body, should_raise",
+    (
+        (HTTP_100_CONTINUE, None, False),
+        (HTTP_101_SWITCHING_PROTOCOLS, None, False),
+        (HTTP_102_PROCESSING, None, False),
+        (HTTP_103_EARLY_HINTS, None, False),
+        (HTTP_204_NO_CONTENT, None, False),
+        (HTTP_100_CONTINUE, "1", True),
+        (HTTP_101_SWITCHING_PROTOCOLS, "1", True),
+        (HTTP_102_PROCESSING, "1", True),
+        (HTTP_103_EARLY_HINTS, "1", True),
+        (HTTP_204_NO_CONTENT, "1", True),
+    ),
+)
+def test_statuses_without_body(status: int, body: Optional[str], should_raise: bool) -> None:
+    @get("/")
+    def handler() -> Response:
+        return Response(content=body, status_code=status)
+
+    with create_test_client(handler) as client:
+        response = client.get("/")
+        if should_raise:
+            assert response.status_code == HTTP_500_INTERNAL_SERVER_ERROR
+        else:
+            assert response.status_code == status
+            assert "content-length" not in response.headers
+
```

Comment:
Maybe add a test for the case that a handler **should** return `None`, but doesn't (and the other way around) when no explicit status code is set (i.e a handler is annotated as `- > None`  without a return code)

Reply:
aight